app/views/vendors/show.html.erb

```diff
@@ -0,0 +1,83 @@
+<div class='nav-two-links'>
+  <%= link_to 'home', '/', class: 'top-nav-button' %>
+  <%= link_to 'all vendors', vendors_path, method: :get, class: "top-nav-button" %>
+</div>
+
+<hr class="nav-hr" />
+
+<h1 class='pg-header'><%= @vendor.name %></h1>
+<br>
+
+<hr class="nav-hr" />
+
+<h2 class='pg-sub-header'>products</h2>
+
+<table class="product_table_list">
+  <% @vendor.products.each do |product| %>
+    <tr>
+      <td><%= product.name.downcase %></td>
+      <td class='button_column'><%= button_to '', edit_vendor_product_path(vendor_id: product.vendor_id, id: product.id ), method: :get, class: 'edit-button' %></td>
+      <td class='button_column'><%= button_to '', product, method: :delete, class: "delete-button" %></td>
+      <td class='button_column'><%= button_to '+ sale', new_product_sale_path(product_id: product.id), method: :get, class: "add-sale-button" %></td>
+    </tr>
+  <% end %>
+</table>
+
+<br>
+
+<div class="one_button_nav">
+  <%= link_to "Add new product", new_vendor_product_path(vendor_id: params[:id]), class: 'nav_button_bottom' %><br><br>
+</div>
+
+
+
+<h2 class='pg-sub-header'>sales</h2>
+
+<div class="two_button_nav_wider">
+  <%= link_to(vendor_path(order: 'asc'), :class => "nav_button_wider") do %>
+    <hr />
+    ascending
+    <hr />
+  <% end %>
+  <%= link_to(vendor_path(order: 'desc'), :class => "nav_button_wider") do %>
+    <hr />
+    descending
+    <hr />
+  <% end %>
+</div>
+
+
+
+<table class='sale_table_list'>
+  <% total_sales = 0 %>
+  <% @sales.each do |sale| %>
+    <% total_sales += sale.amount %>
```

Comment:
It seems like this `total_sales` calculation could be done somewhere else, like the model or controller